Sources/OTT/KalturaOTTPlayer.swift

```diff
@@ -211,3 +215,98 @@ extension KalturaOTTPlayer {
         KPOTTDMSConfigModel.shared.addPartnerConfig(partnerId: partnerId, ovpPartnerId: ovpPartnerId, analyticsUrl: analyticsUrl, ovpServiceUrl: ovpServiceUrl, uiConfId: uiConfId)
     }
 }
+
+// MARK: - Playlists
+
+extension KalturaOTTPlayer {
+    
+    @objc public func loadPlaylist(options: [OTTMediaOptions], callback: @escaping (_ error: NSError?) -> Void) {
+        self.playlistController = nil
+        
+        if options.first?.ks?.isEmpty == false {
+            // TODO: change this logic!
+            sessionProvider.ks = options.first?.ks
+        } else {
+            sessionProvider.ks = playerOptions.ks
+        }
+        
+        let assets: [OTTPlaylistAsset] = options.map { OTTPlaylistAsset(id: $0.assetId,
+                                                                        assetReferenceType: $0.assetReferenceType) }
+        
+        let phoenixPlaylistProvider = PhoenixPlaylistProvider()
+        phoenixPlaylistProvider.set(referrer: KalturaOTTPlayerManager.shared.referrer)
+        phoenixPlaylistProvider.set(sessionProvider: sessionProvider)
+        phoenixPlaylistProvider.set(mediaAssets: assets)
+        
+        phoenixPlaylistProvider.loadPlaylist { [weak self] (playList: PKPlaylist?, error: Error?) in
+            guard let self = self else { return }
+            guard let playList = playList else {
+                if let error = error as? PhoenixMediaProviderError {
+                    callback(error.asNSError)
+                    return
+                }
+                callback(KalturaPlayerError.playlistProviderError.asNSError)
+                return
+            }
+            
+            let controller = KPOTTPlaylistController(playlistConfig: nil,
+                                                     playlist: playList,
+                                                     player: self)
+            
+            controller.originalOTTMediaOptions = options
+            self.playlistController = controller
+            callback(nil)
+        }
+    }
+    
+}
+
+extension KalturaOTTPlayer: EntryLoader {
+    
+    internal func loadMedia(options: MediaOptions, callback: @escaping (_ entry: PKMediaEntry?, _ error: NSError?) -> Void) {
+        guard let options = options as? OTTMediaOptions else {
+            callback(nil, KalturaPlayerError.invalidMediaOptions.asNSError)
+            return
+        }
+        
+        if options.ks?.isEmpty == false {
+            sessionProvider.ks = options.ks
+        } else {
+            sessionProvider.ks = playerOptions.ks
+        }
+        
+        let phoenixMediaProvider = options.mediaProvider()
+        phoenixMediaProvider.set(referrer: KalturaOTTPlayerManager.shared.referrer)
+        phoenixMediaProvider.set(sessionProvider: sessionProvider)
+        
+        phoenixMediaProvider.loadMedia { [weak self] (pkMediaEntry, error) in
```

Comment:
There is no usage of self, this can be removed.
```suggestion
        phoenixMediaProvider.loadMedia { (pkMediaEntry, error) in
```